RecoHGCal/TICL/plugins/PFTICLProducer.cc

```diff
@@ -95,7 +119,30 @@ void PFTICLProducer::produce(edm::StreamID, edm::Event& evt, const edm::EventSet
       reco::TrackRef ref(ticl_cand.trackPtr().id(), int(ticl_cand.trackPtr().key()), &evt.productGetter());
       candidate.setTrackRef(ref);
     }
-    candidate.setTime(ticl_cand.time(), ticl_cand.timeError());
+
+    // HGCAL timing as default values
+    auto time = ticl_cand.time();
+    auto timeE = ticl_cand.timeError();
+
+    if (candidate.charge()) {  // Check MTD timing availability
+      const bool assocQuality = (*trackTimeQualH)[candidate.trackRef()] > timingQualityThreshold_;
+      if (assocQuality) {
+        const auto timeHGC = time;
```

Comment:
These two "HGC" assignments can be moved inside the following `if` block

Reply:
At the moment yes, but when [this comment](https://github.com/cms-sw/cmssw/blob/ec3d88433d2f8d68919298cadde6968b3bb601fa/RecoHGCal/TICL/plugins/PFTICLProducer.cc#L139) will be addressed they will be used also in another `else` of the following `if` block.
So I would leave them where they are.